mono/mini/interp/interp.c

```diff
@@ -435,14 +397,12 @@ get_context (void)
 	ThreadContext *context = (ThreadContext *) mono_native_tls_get_value (thread_context_id);
 	if (context == NULL) {
 		context = g_new0 (ThreadContext, 1);
-		/*
-		 * Initialize data stack.
-		 * There are multiple advantages to this being separate from the frame stack.
-		 * Frame stack can be alloca.
-		 * Frame stack can be perfectly fit (if heap).
-		 * Frame stack can skip GC tracking.
-		 */
-		frame_stack_init (&context->data_stack, 8192);
+		context->stack_start = (guchar*)mono_valloc (0, INTERP_STACK_SIZE, MONO_MMAP_READ | MONO_MMAP_WRITE, MONO_MEM_ACCOUNT_INTERP_STACK);
+		context->stack_pointer = context->stack_start;
```

Comment:
Does this mean that every thread will have 1mb of interp stack trace, which cannot be changed/extended ?

Reply:
Yes. It behaves similarly to a normal thread stack. In the future, sp and vtsp will get unified (meaning we will not have pointers pointing to stack inside the stack space, for valuetypes) and we could have instructions that only do access through constant offsets, from locals / temps that have a known compile type, which we could track to know if it is a scalar or a managed pointer. I think this would make it relatively easy to relocate the entire stack and fixup all pointers from the old stack. To be honest, I don't think it would ever be worth the effort, though.